### PR TITLE
Add basic tests

### PR DESCRIPTION
**Pull request by SeanFeldman**
Basic tests for NServiceBus.AzureFunctions

### CODE REVIEW

**Comment by SeanFeldman** [comment]
TC was missing the environment variable required to read the connection strings. I've added those.

File: src/ServiceBus.Tests/When_message_is_received.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                var bytes = Encoding.UTF8.GetBytes("<HappyDayMessage/>");
+                var message = new Message(bytes);
+                message.UserProperties["NServiceBus.EnclosedMessageTypes"] = typeof(HappyDayMessage).FullName;
+
+                return message;
+            }
+        }
+    }
+
+    public class TestContext
```
Comment (timbussmann) [inline src/ServiceBus.Tests/When_message_is_received.cs]: context, messages and handler classes are usually nested classes in the core acceptance tests to not pollute the namespace too much. Would that make sense too here?

Comment (SeanFeldman) [reply src/ServiceBus.Tests/When_message_is_received.cs]: I'm ok to move it, but not sure what exactly "namespace pollution" in the case of a test project means. This is not a public assembly or shared with customers.

Comment (SeanFeldman) [reply src/ServiceBus.Tests/When_message_is_received.cs]: Thought about it a bit more and here are my thoughts:

1. We can't pollute the namespace as the namespace is defined as a testing namespace. Therefore anything found in the namespace is expected to belong to the tests.
1. These are not Core ATTs. These are tests of the assembly we're not shipping.

As such, I'm good with keeping it as-is.

Comment (SeanFeldman) [reply src/ServiceBus.Tests/When_message_is_received.cs]: Discussed on the call. https://github.com/ParticularLabs/NServiceBus.AzureFunctions/pull/27/commits/e8c6eb5d174c662f84c42684320492a9556edeba

**Comment by SeanFeldman** [comment]
Discussed on the sync call to add an additional test to verify retries are working as expected w/o invoking the infrastructure.

**Pull request merged**